crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -63,50 +76,117 @@ impl MinimapClickEvent {
     }
 }
 
-impl fmt::Debug for MinimapClickEvent {
+impl fmt::Debug for MinimapPressEvent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?} -> {:?}", self.button, self.position)
     }
 }
 
+#[derive(Event)]
+struct MinimapDragEvent {
+    button: MouseButton,
+    position: Vec2,
+}
+
+impl MinimapDragEvent {
+    fn new(button: MouseButton, position: Vec2) -> Self {
+        Self { button, position }
+    }
+
+    fn button(&self) -> MouseButton {
+        self.button
+    }
+
+    /// Position on the map in 2D flat coordinates (these are not minimap
+    /// coordinates).
+    fn position(&self) -> Vec2 {
+        self.position
+    }
+}
+
+#[derive(Resource)]
+struct DraggingButtons(Vec<MouseButton>);
+
 fn click_handler(
     window_query: Query<&Window, With<PrimaryWindow>>,
     mut input_events: EventReader<MouseButtonInput>,
     hud: HudNodes<With<MinimapNode>>,
     bounds: Res<MapBounds>,
-    mut click_events: EventWriter<MinimapClickEvent>,
+    mut dragging: ResMut<DraggingButtons>,
+    mut click_events: EventWriter<MinimapPressEvent>,
```

Comment:
I would finish with the renaming:

* `click_handler()` -> `press_handler()`
* `mut click_events` -> `mut press_events`
* `InteractionSet::ClickHandler` -> `InteractionSet::PressHandler`